src/validation.cpp

```diff
@@ -2655,6 +2656,12 @@ bool Chainstate::ConnectBlock(const CBlock& block, BlockValidationState& state,
                 break;
             }
 
+            if (check_bip54 && !Consensus::CheckSigopsBIP54(tx, view)) {
```

Comment:
84e10433b871bc96a5c10ce3e239adeb28e7af50

Do we want to keep the CheckSigopsBIP54 check in standardness too? (I don't recall what we've done for others)

Reply:
Alternatively we could have the check in `CheckTxInputs`. I did not go for that approach because it involves passing it the deployment status of BIP54. But it may be preferable in order to have a consistent error between block and tx submission (currently it would return a non-standard error for tx submission).

Reply:
soft preference for having the check in a single place, even if it's slight code churn to get there